src/browser/cdp/selectivity/json-utils.ts

```diff
@@ -0,0 +1,158 @@
+            case "zstd":
+                stream = readStream.pipe(
+                    (zlib as unknown as { createZstdDecompress: () => Transform }).createZstdDecompress(),
+                );
+                break;
+            case "br":
+                stream = readStream.pipe(zlib.createBrotliDecompress());
+                break;
+            case "gz":
+                stream = readStream.pipe(zlib.createGunzip());
+                break;
+        }
+
+        stream.setEncoding("utf8");
+
+        stream.on("data", chunk => {
+            fileData += chunk;
+        });
+
+        stream.on("end", () => {
+            resolve(fileData);
+        });
+
+        stream.on("error", err => {
+            reject(new Error(`Couldn't read ${filePath} with ${compression} compression:\n${err}`));
+        });
+    });
+};
+
+const writeCompressedTextFile = (
+    filePath: string,
+    data: string,
+    compression: SelectivityCompressionType,
+): Promise<void> => {
+    return new Promise<void>((resolve, reject) => {
+        const writeStream = fs.createWriteStream(filePath);
+        let stream: Writable = writeStream;
+
+        if (compression === "zstd" && !("createZstdCompress" in zlib)) {
+            reject(new Error("zstd compression is not supported in your version of node.js. Please upgrade to 22"));
+        }
+
+        switch (compression) {
+            case "zstd":
+                stream = (zlib as unknown as { createZstdCompress: () => Transform }).createZstdCompress();
+                stream.pipe(writeStream);
+                break;
+            case "br":
+                stream = zlib.createBrotliCompress();
+                stream.pipe(writeStream);
+                break;
+            case "gz":
+                stream = zlib.createGzip();
+                stream.pipe(writeStream);
+                break;
+        }
+
+        stream.write(data, "utf8", err => {
+            if (err) {
+                reject(new Error(`Couldn't write to ${filePath} with ${compression} compression:\n${err}`));
+            } else {
+                stream.end();
+            }
+        });
+
+        writeStream.on("finish", () => {
+            resolve();
+        });
+
+        writeStream.on("error", err => {
+            reject(new Error(`Couldn't save to ${filePath} with ${compression} compression:\n${err}`));
+        });
+    });
+};
+
+/**
+ * @param jsonBasePath json path without compression suffix
+ * @param preferredCompressionType
+ * @returns decompressed and parsed JSON
+ */
+export const readJsonWithCompression = async <T>(
+    jsonBasePath: string,
+    preferredCompressionType: SelectivityCompressionType,
+    opts?: { defaultValue?: T },
+): Promise<T> => {
+    const { jsonPath, compressionType } = getExistingJsonPathWithCompression(jsonBasePath, preferredCompressionType);
+
+    if (!jsonPath) {
+        if (opts?.defaultValue) {
+            return Promise.resolve(opts.defaultValue);
```

Comment:
Since we check multiple files for existence (".zstd", ".br", ".gz", ""), implement "defaultValue" here, so we dont need to check for existence of all these files before separately